Move all face-connectivity data to its own class : FaceData

adityakpandare opened the pull request:
This set of commits moves all face-data related calls into a new class FaceData. It is a simple class, not a chare-array, which is created per chare-worker in Partitioner::createWorkers(). This is done so that Discretization will not hold all this data unnecessarily, viz. for MatCG and DiagCG, where it's not needed. It will be computed only when DG discretization is chosen.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/quinoacomputing/quinoa/198)
<!-- Reviewable:end -->

jbakosi commented on src/Inciter/DG.C:
```diff
@@ -39,6 +41,7 @@ DG::DG( const CProxy_Discretization& disc, const tk::CProxy_Solver& solver ) :
 {
   // Signal the runtime system that the workers have been created
   solver.ckLocalBranch()->created();
+  (void)fd;
```
For future reference: this could also be done using the macro `IGNORE`, defined in `Base/Macro.h`. It does the same as you did, but using C++'s `static_cast`.

adityakpandare replied:
Okay, made this change anyway.

jbakosi commented on src/Inciter/FaceData.C:
```diff
@@ -0,0 +1,67 @@
+// *****************************************************************************
+/*!
+  \file      src/Inciter/FaceData.C
+  \copyright 2012-2015, J. Bakosi, 2016-2018, Los Alamos National Security, LLC.
```
For any new file please use:
```
\copyright 2016-2018, Los Alamos National Security, LLC.
```

jbakosi commented on src/Inciter/FaceData.C:
```diff
@@ -0,0 +1,67 @@
+//!   is in start contrast with "as-in-file" ordering, which is the ordering of
+//!   the mesh nodes as it is stored in the file from which the mesh is read in.
+//!   The as-in-file ordering is highly non-contiguous across the distributed
+//!   problem.
```
This `\details` is not needed here.

jbakosi commented on src/Inciter/FaceData.C:
```diff
@@ -0,0 +1,67 @@
+// *****************************************************************************
+//  Constructor
+//! \param[in] conn Vector of mesh element connectivity owned (global IDs)
+//!   mesh chunk we operate on
```
Please add `\param` doxygen comments for the rest of the constructor arguments as well.

jbakosi commented on src/Inciter/FaceData.C:
```diff
@@ -0,0 +1,67 @@
+// *****************************************************************************
+
+#include "Reorder.h"
+#include "Vector.h"
```
Including `Vector.h` may not be necessary.

jbakosi commented on src/Inciter/FaceData.h:
```diff
@@ -0,0 +1,124 @@
+namespace tk {
+  class ExodusIIMeshWriter;
+  class RootMeshWriter;
+}
```
None of
```
#include "UnsMesh.h"
￼#include "Inciter/InputDeck/InputDeck.h"
￼
￼namespace tk {
￼  class ExodusIIMeshWriter;
￼  class RootMeshWriter;
￼}
```
may be necessary.

jbakosi commented on src/Inciter/FaceData.h:
```diff
@@ -0,0 +1,124 @@
+
+namespace inciter {
+
+extern ctr::InputDeck g_inputdeck;
```
This may not be necessary either.

jbakosi commented on src/Inciter/FaceData.h:
```diff
@@ -0,0 +1,124 @@
+    std::unordered_map< std::size_t, std::size_t >& Lid() { return m_lid; }
+
+    const std::vector< std::size_t >& Inpoel() const { return m_inpoel; }
+    std::vector< std::size_t >& Inpoel() { return m_inpoel; }
```
Remember that `m_el` (and thus all of `m_inpoel`, `m_gid`, and `m_lid`) is a duplicate here. Accordingly, let's not even store these in `FaceData`: create it as a local variable in the constructor, use them there and it will be correctly thrown away at the end of the scope. Thus these accessors, `Gid`, `Lid`, and `Inpoel`, should also go away.

jbakosi commented on src/Inciter/FaceData.h:
```diff
@@ -0,0 +1,124 @@
+        m_inpoel = std::get< 0 >( m_el );
+        m_gid = std::get< 1 >( m_el );
+        m_lid = std::get< 2 >( m_el );
+      }
```
These will also go away: `m_el`, and the `isUnPacking` block.

jbakosi commented on src/Inciter/FaceData.h:
```diff
@@ -0,0 +1,124 @@
+    std::vector< std::size_t > m_gid = std::get< 1 >( m_el );
+    //! \brief Alias to local node ids associated to the global ones of owned
+    //!    elements in m_el
+    std::unordered_map< std::size_t, std::size_t > m_lid = std::get< 2 >( m_el );
```
As well as these will also go away: `m_el`, `m_gid`, `m_lid`, m_inpoel`.

jbakosi commented on src/Inciter/Partitioner.C:
```diff
@@ -1232,9 +1232,9 @@ Partitioner::createDiscWorkers()
-  tk::destroy( m_chinpoel );
+  //// Free storage of global mesh node ids associated to chares owned as it
+  //// is no longer needed after creating the workers.
+  //tk::destroy( m_chinpoel );
```
Let's just delete this together with its comments.

jbakosi commented on src/Inciter/Scheme.h:
```diff
@@ -218,7 +218,7 @@ class Scheme : public SchemeBase {
     template< typename Op, typename... Args, typename std::enable_if<
       std::is_same< Op, tag::elem >::value, int >::type = 0,
-      typename std::enable_if< sizeof...(Args) == 11, int >::type = 0 >
+      typename std::enable_if< sizeof...(Args) == 8, int >::type = 0 >
```
Just a note: all of this will go away once https://github.com/quinoacomputing/quinoa/pull/194 is merged.

jbakosi commented on src/Inciter/Scheme.h:
```diff
@@ -231,7 +231,7 @@ class Scheme : public SchemeBase {
     template< typename Op, typename... Args, typename std::enable_if<
       std::is_same< Op, tag::elem >::value, int >::type = 1,
-      typename std::enable_if< sizeof...(Args) == 12, int >::type = 0 >
+      typename std::enable_if< sizeof...(Args) == 9, int >::type = 0 >
```
Same here.

jbakosi commented on src/UnitTest/CMakeLists.txt:
```diff
@@ -25,6 +25,7 @@ add_dependencies("UnitTest" "unittestCharmModule")
 
 # Add extra dependencies of UnitTest on those CharmModules that are required for
 # testing Inciter.
+add_dependencies("UnitTest" "facedataCharmModule")
```
This now does not exist, correct?

jbakosi commented:
Reviewed 14 of 21 files at r1, 1 of 4 files at r2, 6 of 6 files at r3.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[src/Inciter/DG.C, line 44 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BRRXtqRBIRFp3f-HB:-L4BRRXtqRBIRFp3f-HC:b5gr1kp) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/DG.C#L44)):*
> ```
>   // Signal the runtime system that the workers have been created
>   solver.ckLocalBranch()->created();
>   (void)fd;
> ```

For future reference: this could also be done using the macro `IGNORE`, defined in `Base/Macro.h`. It does the same as you did, but using C++'s`static_cast`.

---

*[src/Inciter/FaceData.C, line 4 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BSu97VUMV6p3aPc8Q:-L4BSu97VUMV6p3aPc8R:bbjcaor) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.C#L4)):*
> ```
> /*!
>   \file      src/Inciter/FaceData.C
>   \copyright 2012-2015, J. Bakosi, 2016-2018, Los Alamos National Security, LLC.
> ```

For any new file please use:
```
\copyright 2016-2018, Los Alamos National Security, LLC.
```

---

*[src/Inciter/FaceData.C, line 11 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BTaujtM83qPiPPYDE:-L4BTaukzx-F_Z7vte_t:b-lj81k9) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.C#L11)):*
> ```
> 
> #include "Reorder.h"
> #include "Vector.h"
> ```

Including `Vector.h` may not be necessary.

---

*[src/Inciter/FaceData.C, line 32 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BTRhVkZOUgdBH61Hq:-L4BTRhVkZOUgdBH61Hr:by2ncg3) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.C#L32)):*
> ```
> //  Constructor
> //! \param[in] conn Vector of mesh element connectivity owned (global IDs)
> //!   mesh chunk we operate on
> ```

Please add `\param` doxygen comments for the rest of the constructor arguments as well.

---

*[src/Inciter/FaceData.C, line 40 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BT5Ql4CLaSdJR-iFR:-L4BT5Ql4CLaSdJR-iFS:b-4oev9v) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.C#L40)):*
> ```
> //!   the mesh nodes as it is stored in the file from which the mesh is read in.
> //!   The as-in-file ordering is highly non-contiguous across the distributed
> //!   problem.
> ```

This `\details` is not needed here.

---

*[src/Inciter/FaceData.h, line 18 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BTs09tv5kd1_yXi-Q:-L4BTs09tv5kd1_yXi-R:b-qxj556) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.h#L18)):*
> ```objc
>   class ExodusIIMeshWriter;
>   class RootMeshWriter;
> }
> ```

None of
```
#include "UnsMesh.h"
￼#include "Inciter/InputDeck/InputDeck.h"
￼
￼namespace tk {
￼  class ExodusIIMeshWriter;
￼  class RootMeshWriter;
￼}
```
may be necessary.

---

*[src/Inciter/FaceData.h, line 22 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BTzN--745d23Tpk53:-L4BTzN--745d23Tpk54:bwfmbxc) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.h#L22)):*
> ```objc
> namespace inciter {
> 
> extern ctr::InputDeck g_inputdeck;
> ```

This may not be necessary either.

---

*[src/Inciter/FaceData.h, line 51 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BU2V9Qz1gJZe7Gxmg:-L4BU2V9Qz1gJZe7Gxmh:b8spu7n) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.h#L51)):*
> ```objc
> 
>     const std::vector< std::size_t >& Inpoel() const { return m_inpoel; }
>     std::vector< std::size_t >& Inpoel() { return m_inpoel; }
> ```

Remember that `m_el` (and thus all of `m_inpoel`, `m_gid`, and `m_lid`) is a duplicate here. Accordingly, let's not even store these in `FaceData`: create it as a local variable in the constructor, use them there and it will be correctly thrown away at the end of the scope. Thus these accessors, `Gid`, `Lid`, and `Inpoel`, should also go away.

---

*[src/Inciter/FaceData.h, line 72 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BUG2g1DMej0Yd3k18:-L4BUG2g1DMej0Yd3k19:bjuxeh) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.h#L72)):*
> ```objc
>         m_gid = std::get< 1 >( m_el );
>         m_lid = std::get< 2 >( m_el );
>       }
> ```

These will also go away: `m_el`, and the `isUnPacking` block.

---

*[src/Inciter/FaceData.h, line 103 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BULwInY-StrpfZ9Tu:-L4BULwInY-StrpfZ9Tv:b-4wvolm) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.h#L103)):*
> ```objc
>     //! \brief Alias to local node ids associated to the global ones of owned
>     //!    elements in m_el
>     std::unordered_map< std::size_t, std::size_t > m_lid = std::get< 2 >( m_el );
> ```

As well as these will also go away: `m_el`, `m_gid`, `m_lid`, `m_inpoel`.

---

*[src/Inciter/Partitioner.C, line 1237 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BUbBw6McWJsb2gVdm:-L4BUbBw6McWJsb2gVdn:bqg5xcb) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/Partitioner.C#L1237)):*
> ```
>   //// Free storage of global mesh node ids associated to chares owned as it
>   //// is no longer needed after creating the workers.
>   //tk::destroy( m_chinpoel );
> ```

Let's just delete this together with its comments.

---

*[src/Inciter/Scheme.h, line 221 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BUo6xKHyqLD6p2MGD:-L4BUo6xKHyqLD6p2MGE:b-he74ya) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/Scheme.h#L221)):*
> ```objc
>     template< typename Op, typename... Args, typename std::enable_if<
>       std::is_same< Op, tag::elem >::value, int >::type = 0,
>       typename std::enable_if< sizeof...(Args) == 8, int >::type = 0 >
> ```

Just a note: all of this will go away once #194 is merged.

---

*[src/Inciter/Scheme.h, line 234 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BUrxgK8qiO3grCM-J:-L4BUrxgK8qiO3grCM-K:brhumzt) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/Scheme.h#L234)):*
> ```objc
>     template< typename Op, typename... Args, typename std::enable_if<
>       std::is_same< Op, tag::elem >::value, int >::type = 1,
>       typename std::enable_if< sizeof...(Args) == 9, int >::type = 0 >
> ```

Same here.

---

*[src/UnitTest/CMakeLists.txt, line 28 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BUxpzjhP5fwq7qQxP:-L4BUxpzjhP5fwq7qQxQ:bm1a51v) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/UnitTest/CMakeLists.txt#L28)):*
> ```CMake
> # Add extra dependencies of UnitTest on those CharmModules that are required for
> # testing Inciter.
> add_dependencies("UnitTest" "facedataCharmModule")
> ```

This now does not exist, correct?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198)*
<!-- Sent from Reviewable.io -->

jbakosi commented:
Very nice work! Please address the minor comments, push the changes, wait for travis, then if all green, merge to develop. (Don't forget to sign the merge commit before push.) This is in a great shape now. Thanks a lot!

---

Review status: all files reviewed at latest revision, 13 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-:-L4BXJCO0eI6_KHdA1Pw:b7cxr0o)*
<!-- Sent from Reviewable.io -->

adityakpandare commented:
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[src/Inciter/DG.C, line 44 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BRRXtqRBIRFp3f-HB:-L4BaIlZMHFUPOwzUp3u:b-wk2vs4) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/DG.C#L44)):*
<details><summary><i>Previously, jbakosi (Jozsef Bakosi) wrote…</i></summary><blockquote>

For future reference: this could also be done using the macro `IGNORE`, defined in `Base/Macro.h`. It does the same as you did, but using C++'s`static_cast`.
</blockquote></details>

Took care of this

---

*[src/Inciter/FaceData.C, line 4 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BSu97VUMV6p3aPc8Q:-L4BaRoO7BNJIPNsuPgR:b-896fix) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.C#L4)):*
<details><summary><i>Previously, jbakosi (Jozsef Bakosi) wrote…</i></summary><blockquote>

For any new file please use:
```
\copyright 2016-2018, Los Alamos National Security, LLC.
```
</blockquote></details>

Done.

---

*[src/Inciter/FaceData.C, line 11 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BTaujtM83qPiPPYDE:-L4BaUsDyfKa90J2JKGv:b-896fix) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.C#L11)):*
<details><summary><i>Previously, jbakosi (Jozsef Bakosi) wrote…</i></summary><blockquote>

Including `Vector.h` may not be necessary.
</blockquote></details>

Done.

---

*[src/Inciter/FaceData.C, line 40 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BT5Ql4CLaSdJR-iFR:-L4BaW6bfTxAo0t13oQ7:ba0iqhd) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.C#L40)):*
<details><summary><i>Previously, jbakosi (Jozsef Bakosi) wrote…</i></summary><blockquote>

This `\details` is not needed here.
</blockquote></details>

Corrected with the right details

---

*[src/Inciter/FaceData.h, line 18 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BTs09tv5kd1_yXi-Q:-L4Ba_2vdfC65ntjWRyb:b-80bnvp) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.h#L18)):*
<details><summary><i>Previously, jbakosi (Jozsef Bakosi) wrote…</i></summary><blockquote>

None of
```
#include "UnsMesh.h"
￼#include "Inciter/InputDeck/InputDeck.h"
￼
￼namespace tk {
￼  class ExodusIIMeshWriter;
￼  class RootMeshWriter;
￼}
```
may be necessary.

</blockquote></details>

True.

---

*[src/Inciter/FaceData.h, line 51 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BU2V9Qz1gJZe7Gxmg:-L4BaeNSHj93rt4GuNYt:bjjx6f0) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.h#L51)):*
<details><summary><i>Previously, jbakosi (Jozsef Bakosi) wrote…</i></summary><blockquote>

Remember that `m_el` (and thus all of `m_inpoel`, `m_gid`, and `m_lid`) is a duplicate here. Accordingly, let's not even store these in `FaceData`: create it as a local variable in the constructor, use them there and it will be correctly thrown away at the end of the scope. Thus these accessors, `Gid`, `Lid`, and `Inpoel`, should also go away.
</blockquote></details>

Yep. Removed these.

---

*[src/Inciter/Partitioner.C, line 1237 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BUbBw6McWJsb2gVdm:-L4BajL1D-QbcFowxUra:b-837i3v) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/Partitioner.C#L1237)):*
<details><summary><i>Previously, jbakosi (Jozsef Bakosi) wrote…</i></summary><blockquote>

Let's just delete this together with its comments.
</blockquote></details>

Okay.

---

*[src/Inciter/Scheme.h, line 221 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BUo6xKHyqLD6p2MGD:-L4BakwsX4yRB6NTfyx2:bx8ijcy) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/Scheme.h#L221)):*
<details><summary><i>Previously, jbakosi (Jozsef Bakosi) wrote…</i></summary><blockquote>

Just a note: all of this will go away once #194 is merged.
</blockquote></details>

How will this go away?

---

*[src/UnitTest/CMakeLists.txt, line 28 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BUxpzjhP5fwq7qQxP:-L4BasisGQga1rxoIHVk:bd7t2it) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/UnitTest/CMakeLists.txt#L28)):*
<details><summary><i>Previously, jbakosi (Jozsef Bakosi) wrote…</i></summary><blockquote>

This now does not exist, correct?
</blockquote></details>

Correct. Removed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198)*
<!-- Sent from Reviewable.io -->

adityakpandare commented:
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---

*[src/Inciter/FaceData.C, line 32 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BTRhVkZOUgdBH61Hq:-L4BbFyAATidumSQa24E:b-896fix) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.C#L32)):*
<details><summary><i>Previously, jbakosi (Jozsef Bakosi) wrote…</i></summary><blockquote>

Please add `\param` doxygen comments for the rest of the constructor arguments as well.
</blockquote></details>

Done.

---

*[src/Inciter/FaceData.h, line 22 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BTzN--745d23Tpk53:-L4BbLH-S0dXKoRagh76:b-896fix) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.h#L22)):*
<details><summary><i>Previously, jbakosi (Jozsef Bakosi) wrote…</i></summary><blockquote>

This may not be necessary either.
</blockquote></details>

Done.

---

*[src/Inciter/FaceData.h, line 72 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BUG2g1DMej0Yd3k18:-L4BbI7d7Dx7S10HLXsr:b-896fix) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.h#L72)):*
<details><summary><i>Previously, jbakosi (Jozsef Bakosi) wrote…</i></summary><blockquote>

These will also go away: `m_el`, and the `isUnPacking` block.
</blockquote></details>

Done.

---

*[src/Inciter/FaceData.h, line 103 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BULwInY-StrpfZ9Tu:-L4BbBsNxSBvmYNcCE0i:b-896fix) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/FaceData.h#L103)):*
<details><summary><i>Previously, jbakosi (Jozsef Bakosi) wrote…</i></summary><blockquote>

As well as these will also go away: `m_el`, `m_gid`, `m_lid`, `m_inpoel`.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198)*
<!-- Sent from Reviewable.io -->

jbakosi commented:
Review status: 14 of 20 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/Inciter/Scheme.h, line 221 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BUo6xKHyqLD6p2MGD:-L4BbmiuvTrHHk1ZbXb_:bewfyg6) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/Scheme.h#L221)):*
<details><summary><i>Previously, adityakpandare (Aditya Pandare) wrote…</i></summary><blockquote>

How will this go away?
</blockquote></details>

These types of changes (changing the number of the arguments) will no longer be needed because #194 will remove them.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198)*
<!-- Sent from Reviewable.io -->

adityakpandare commented:
Thanks Jozsef. I'll be more careful with comments in the codes (for doxygen) in the future.

---

Review status: 14 of 20 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-:-L4Bc90BAxWcYqBZYjDi:bmtlr5a)*
<!-- Sent from Reviewable.io -->

jbakosi commented:
Whoa! I think I love this thing!

---

Reviewed 6 of 7 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-:-L4BclM3EeF-rQwoxJcY:b-rz7n3q)*
<!-- Sent from Reviewable.io -->

adityakpandare commented:
:smile: It is pretty neat!

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---

*[src/Inciter/Scheme.h, line 221 at r3](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-L4BUo6xKHyqLD6p2MGD:-L4BdhmG-V8fBI1fxIxx:b-gvx6jy) ([raw file](https://github.com/quinoacomputing/quinoa/blob/10d2da700e151de3d68ddd09d73bb821ca428fb8/src/Inciter/Scheme.h#L221)):*
<details><summary><i>Previously, jbakosi (Jozsef Bakosi) wrote…</i></summary><blockquote>

These types of changes (changing the number of the arguments) will no longer be needed because #194 will remove them.
</blockquote></details>

That's right. now I get it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quinoacomputing/quinoa/198#-:-L4Bd74rvZ6wwlJF5V0U:b-maaejd)*
<!-- Sent from Reviewable.io -->
